huggingface_pipelines/pipeline_config.py

```diff
@@ -0,0 +1,179 @@
+from abc import ABC
+from dataclasses import dataclass, replace
+from typing import List
+
+
+@dataclass
+class PipelineConfig(ABC):
+    """
+    Abstract base class for pipeline configurations.
+
+    Attributes:
+        dataset_name (str): The name of the HuggingFace dataset to be used.
+        dataset_split (str): The dataset split to be used (e.g., 'train', 'test', 'validation').
+        batch_size (int): The batch size to be used for processing.
+        device (str): The device to use for inference (e.g., 'cpu', 'cuda').
+        pipeline_type (str): The type of pipeline (e.g., 'text', 'audio').
+        num_shards (int): The number of shards to split the dataset into. Defaults to 1.
+        shard_id (int): The ID of the shard to process. Defaults to 0.
+        cache_to_arrow (bool): Whether to cache results to Arrow format. Defaults to False.
+        output_file_name (str): The base name of the file where results will be saved. Defaults to "results".
+        take (int): The number of batches to take for processing. Defaults to -1 (process all).
+
+    """
+    dataset_name: str
+    columns: List[str]
+    dataset_split: str = "test"
+    batch_size: int = 5
+    device: str = "cpu"
+    num_shards: int = 1
+    shard_id: int = 0
+    cache_to_arrow: bool = False
+    output_file_name: str = "results"
+    take: int = -1
+
+    def with_dataset_name(self, dataset_name: str):
+        return replace(self, dataset_name=dataset_name)
+
+    def with_dataset_split(self, dataset_split: str):
+        return replace(self, dataset_split=dataset_split)
+
+    def with_batch_size(self, batch_size: int):
+        return replace(self, batch_size=batch_size)
+
+    def with_device(self, device: str):
+        return replace(self, device=device)
+
+    def with_pipeline_type(self, pipeline_type: str):
+        return replace(self, pipeline_type=pipeline_type)
+
+    def with_num_shards(self, num_shards: int):
+        return replace(self, num_shards=num_shards)
+
+    def with_shard_id(self, shard_id: int):
+        return replace(self, shard_id=shard_id)
+
+    def with_cache_to_arrow(self, cache_to_arrow: bool):
+        return replace(self, cache_to_arrow=cache_to_arrow)
+
+    def with_output_file_name(self, output_file_name: str):
+        return replace(self, output_file_name=output_file_name)
+
+    def with_columns(self, columns: List[str]):
+        return replace(self, columns=columns)
+
+    def with_take(self, take: int):
+        return replace(self, take=take)
+
+
+@dataclass
+class TextToEmbeddingPipelineConfig(PipelineConfig):
```

Comment:
Let's place the corresponding Configs to the python modules where their pipelines are implemented and not all together here